### PR TITLE
Drag between nodes

### LINKED ISSUE #31: Drag nodes between nodes
After #26 is merged you can append nodes or drag a node on top of existing node to push the existing node down. It would be less confusing if a node can be dragged between to existing nodes or above the first or below the last node.

Generated from https://github.com/i-VRESSE/workflow-builder/pull/26#pullrequestreview-876091884

### LINKED ISSUE #94: Rename workflow node properties
The IWorkflowNode type uses `id` property to store the name of the catalog node and `code` as a uniq id which is required for dnd.
See https://github.com/i-VRESSE/workflow-builder/blob/524cb735abcd0bf88f8994c9e6f5395d1526272f/src/types.ts#L53-L56

It would better to rename those properties to 
* `id` -> `type`
* `code` -> `id`

### PR DESCRIPTION
**Pull request by sverhoeven**
Fixes #31 and fixes #94

TODO
* [x] drag catalog node,
  * [x] now when dropped anywhere will append to wf
  * [x] no clear dropzone where to drop
  * ~[ ] when drag is started the catalog node is removed from list~ I can live with it because after drop it reappears.
* [x] dragging wf node sometime generates random new node
* [x] only show drag and x when hover over node
* [x] when moving wf node the selected node is not moved with it
* [x] drag start removes catalog node from list, it should make copy
* [x] no animation where catalog node is dropped

### CODE REVIEW

**Comment by bvreede** [comment]
Each node now takes up more space. Is it possible to reduce this, make them smaller?

| `main`: | `dnd`: |
|---|---|
|| |

**Comment by sverhoeven** [comment]
Spin off https://github.com/i-VRESSE/workflow-builder/issues/94 from pair reviewing session.

**Comment by sverhoeven** [comment]
> Each node now takes up more space. Is it possible to reduce this, make them smaller?
> 

I have reduced the space between nodes from 10px to 4px.

**Comment by sverhoeven** [comment]
VS Code was really helpful doing #94 so I merged the fix for #94 into this PR.

**Comment by bvreede** [comment]
A question, while reviewing: 

In the current implementation, when picking an element, dragging it slightly, then moving it back to its original position, it is seen as 'dropped' and the element is appended to the list. This to me is confusing: once an item is dragged, there is no limits on the page where it needs to be dropped, and the user can't decide to "put it back".

In the TODO above, you write that there need be 'no clear dropzone where to drop'. I'm not sure if this implementation is therefore intentionally designed as such?

File: src/VisualNode.tsx [hunk 1]
```diff
@@ -1,98 +1,73 @@
...
+import { CSS } from '@dnd-kit/utilities'
+import { useSortable } from '@dnd-kit/sortable'
+import { useDraggingWorkflowNodeState, useSelectNodeIndex, useWorkflow } from './store'
+import classes from './VisualNode.module.css'
 
 interface IProp {
   id: string
   index: number
+  code: string
 }
```
Comment (bvreede) [inline src/VisualNode.tsx:11]: id/code was adjusted to type/ID (in 4b445b1 and 1d8dccc) — should this be included? 
(+ other instances in this file)

**Review by bvreede — CHANGES_REQUESTED** [review]
Excellent addition :) this looks very spiffy!

The one behavior I personally do not like (as mentioned in my previous comment) is the indiscriminate dropping. If this is not an intentional design, I suggest setting some boundaries for dropping (I noticed there were no [`Droppable`](https://docs.dndkit.com/introduction/getting-started#droppable) components? Would this help?).

File: src/VisualNode.tsx (hunk 1 above)
Comment (sverhoeven) [reply src/VisualNode.tsx]: Good catch, that component does not use IWorkflowNode type so I missed it.

**Comment by sverhoeven** [comment]
> A question, while reviewing:
> 
> In the current implementation, when picking an element, dragging it slightly, then moving it back to its original position, it is seen as 'dropped' and the element is appended to the list. This to me is confusing: once an item is dragged, there is no limits on the page where it needs to be dropped, and the user can't decide to "put it back".
> 
> In the TODO above, you write that there need be 'no clear dropzone where to drop'. I'm not sure if this implementation is therefore intentionally designed as such?

Dropping a catalog node outside the drop zone now does not add it anymore.
While dragging a node I show the enlarged dropzone.

@bvreede is this better?

**Review by bvreede — APPROVED** [review]
Yass! Nice!

**Comment by sverhoeven** [comment]
Thanks for reviewing and making it even nicer.

**Pull request merged**